From 385c70d9692b11ae626594b9be0626856dd54d50 Mon Sep 17 00:00:00 2001
From: "Yang, Dong" <dong.yang@intel.com>
Date: Fri, 15 May 2015 17:08:41 +0800
Subject: [PATCH 398/462] fix the css shading table memory leak

the css driver need to free shading table during stopping

backtrace:
[ffffffff819d936e] kmemleak_alloc+0x4e/0xb0
[ffffffff8119bbf2] __kmalloc+0xd2/0x4d0
[ffffffffa02be34c] atomisp_kernel_malloc+0x1c/0x30
[ffffffffa02be379] atomisp_kernel_zalloc+0x19/0x50
[ffffffffa02e2d6b] sh_css_malloc+0x3b/0x50
[ffffffffa02ef839] ia_css_shading_table_alloc+0x39/0x110
[ffffffffa02ded6e] atomisp_css_shading_table_alloc+0xe/0x10
[ffffffffa02c1a22] atomisp_cp_lsc_table+0xa2/0x270
[ffffffffa02c3297] atomisp_set_parameters+0x117/0x290
[ffffffffa02b6c4c] atomisp_vidioc_default+0x91c/0xc70
[ffffffff8170fb17] __video_do_ioctl+0x197/0x330
[ffffffff8170f5a0] video_usercopy+0x240/0x600
[ffffffff8170f975] video_ioctl2+0x15/0x20
[ffffffff8170b7fb] v4l2_ioctl+0x13b/0x150
[ffffffffa02bbb55] atomisp_do_compat_ioctl+0x585/0x2040
[ffffffffa02bd6cd] atomisp_compat_ioctl32+0xbd/0x630

Change-Id: Ie8292504a242b06371ffd4fc43b11bd4e449f6fc
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-35973
Signed-off-by: Yang, Dong <dong.yang@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/56468
Reviewed-by: Yunliang Ding <yunliang.ding@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Approver: Tian Shu Qiu <tian.shu.qiu@intel.com>
Build: IMIN (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c     | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
index 1cb8ef2..d690757 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
@@ -2406,6 +2406,7 @@ int atomisp_css_stop(struct atomisp_sub_device *asd,
 	atomisp_flush_params_queue(&asd->video_out_vf);
 	atomisp_flush_params_queue(&asd->video_out_preview);
 	atomisp_flush_params_queue(&asd->video_out_video_capture);
+	atomisp_free_css_parameters(&asd->params.css_param);
 	memset(&asd->params.css_param, 0, sizeof(asd->params.css_param));
 	return 0;
 }
-- 
1.9.1

